src/plugins/files/public/components/util/image_metadata.ts

```diff
@@ -78,3 +78,37 @@ export async function getImageMetadata(file: File | Blob): Promise<undefined | F
 }
 
 export type ImageMetadataFactory = typeof getImageMetadata;
+
+export function getBlurhashSrc({
+  width,
+  height,
+  hash,
+}: {
+  width: number;
+  height: number;
+  hash: string;
+}): Promise<string> {
+  const canvas = document.createElement('canvas');
+  const { width: blurWidth, height: blurHeight } = fitToBox(width, height);
+  canvas.width = blurWidth;
+  canvas.height = blurHeight;
+
+  const ctx = canvas.getContext('2d')!;
+  const imageData = ctx.createImageData(blurWidth, blurHeight);
+  imageData.data.set(bh.decode(hash, blurWidth, blurHeight));
+  ctx.putImageData(imageData, 0, 0);
+
+  return new Promise((resolve) => {
+    // On this point canvas contains a downsized blurred image
```

Comment:
This is the part I don't like but couldn't find the workaround. 
To make the blurred image sizing behave just like the original image we must have the blurred image stretched back to the original size. The problem is that I couldn't find a way to do this synchronously, so now there is a moment in time where image component doesn't render nor the blurred version, nor the  src version

Reply:
Maybe something like below. [`image`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage) param can be another canvas.

```
ctx2.drawImage(ctx, 0, 0, blurWidth, blurHeight, 0, 0, width, height);
```

Reply:
@vadimkibana, the idea with the second canvas works. thanks

I was trying it before like this: 

`ctx2.drawImage(ctx1, 0, 0, width, height);`

But it actually should be like this (drawImage can receive canvas element, not ctx): 

```
ctx2.drawImage(canvas1, 0, 0, width, height);
```


> Actually, can we just return canvas.toDataURL() once we have the first canvas set up and skip the async image loading step? 

I needed the second step (the image loading or the second canvas) to resize the blurred image back to the original size, so it behaves the same as the original image